File: solutions/0020_valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack <char> stk;
        for(char c :s){
            if(stk.empty()){
                stk.push(c);
                continue;
            }
            string temp = string(1,stk.top())+string(1,c);
            if(temp == "()" || temp=="[]" || temp=="{}"){
                stk.pop();
            }else{
                stk.push(c);
            }
        }
        return stk.empty();
    }
};

File: solutions/0020_valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        temp = int(len(s)/2)
        for i in range(temp):
            s = s.replace("()","")
            s = s.replace("{}","")
            s = s.replace("[]","")
        if s=="":
            return True
        return False